src/d3a/models/strategy/predefined_load.py

```diff
@@ -99,10 +99,9 @@ def _update_energy_requirement(self):
                             for day in range(self.area.config.sim_duration.days + 1)}
         for slot_time in generate_market_slot_list(area=self.area):
             if self._allowed_operating_hours(slot_time.hour):
-                self.energy_requirement_Wh[slot_time] = \
-                    self.load_profile[slot_time] * 1000
-                self.state.desired_energy_Wh[slot_time] = \
-                    self.load_profile[slot_time] * 1000
+                self.energy_requirement_Wh[slot_time] = self.load_profile[slot_time] * 1000
+                self.state.desired_energy_Wh[slot_time] = self.load_profile[slot_time] * 1000
+                self.state.total_energy_demanded_wh += self.load_profile[slot_time] * 1000
```

Comment:
Does not hurt, but I think that you could avoid adding the total_energy_demanded_wh to the state class, by adding a property to the state class that accumulates the values of the desired_energy_Wh dict. Optional, if you prefer it this way keep it, should not hurt.

Reply:
Well the issue here is `class LoadState` is used by both `LoadHoursStrategy` and `DefinedLoadStrategy`. 

- For LoadHoursStrategy, `total_energy_demanded_wh = avg_power_W * hrs_per_day`
- For DefinedLoadStrategy, `total_energy_demanded_wh` is accumulated via iterating over load_profiles.

The reason im skeptical of using `desired_energy_Wh` to accumulate `total_energy_demanded_wh` is due to its different  behaviour in both strategy. In LoadHoursStrategy, it is populated for all the hrs_of_day due which it gets accumulated more than the required operating hours.

We can also discuss offline, if it is unclear.

Reply:
All clear, agreed. Thanks